PrBuild/index.js

```diff
@@ -11,165 +11,61 @@ module.exports = function (context, req) {
         const sha = req.body.pull_request.head.sha;
         const target_branch = req.body.pull_request.base.ref;
         const pull_request = req.body.pull_request.id;
-        
-        const appcenter_token = process.env["APP_CENTER_TOKEN"];
-        const github_token = process.env["GITHUB_TOKEN"];
-        const config = require('./config.json');
-        const { repo_owner, repo_name, appcenter_owner, appcenter_app, branch_template, branch_regex, appcenter_owner_type } = config;
-
-        const repo_path = `${repo_owner}/${repo_name}`;
-
-        let message = "";
 
-        if(head_repo == repo_path) {
+        const appcenter_token = process.env['APP_CENTER_TOKEN'];
+        const github_token = process.env['GITHUB_TOKEN'];
 
-            if(action === "opened" || action === "synchronize") {
+        const config = require('./config.json');
+        const repo_config = config.repos.filter((repo) => {
+            return head_repo === `${repo.repo_owner}/${repo.repo_name}`;
+        });
+
+        if (repo_config.length === 1) {
+            const { repo_owner, repo_name, appcenter_owner, appcenter_app, branch_template, appcenter_owner_type } = repo_config[0];
+            const repo_path = `${repo_owner}/${repo_name}`;
+            if (action === 'opened' || action === 'synchronize') {
                 context.log(`PR #${pull_request} was ${action} on '${branch}' trying to merge into '${target_branch}'...`);
-                GetBuildConfiguration(branch, appcenter_token, appcenter_owner, appcenter_app)
-                .then(branch_config => {
-                    if(!branch_config) {
-                        GetBuildConfiguration(branch_template, appcenter_token, appcenter_owner, appcenter_app)
-                        .then(branch_config => {
-                            CreatePrBuildConfiguration(branch_config, branch, appcenter_token, appcenter_owner, appcenter_app)
-                            .then(() => {
-                                StartPrBuild(branch, sha, appcenter_token, appcenter_owner, appcenter_app, appcenter_owner, appcenter_owner_type, appcenter_app)
+                appCenterRequests.getBuildConfiguration(branch, appcenter_token, appcenter_owner, appcenter_app)
```

Comment:
Just as a general comment for all calls to App Center. We need to spend a little time testing what happens when a 401 happens because errors weren't being caught and the context returned correctly prior.

Reply:
Absolutely agreed, there was an initial version of code refactoring, so I've already added exceptions handling for request, going to publish later this week.